--- ffms2-2.20/src/core/matroskareader.cpp~	2014-06-20 17:09:10.000000000 +0200
+++ ffms2-2.20/src/core/matroskareader.cpp	2018-04-29 09:42:31.607973790 +0200
@@ -301,8 +301,8 @@
 }
 
 void MatroskaReaderContext::Append(const void *Data, size_t Length) {
-	if (BufferSize < Length + FrameSize + FF_INPUT_BUFFER_PADDING_SIZE) {
-		size_t NewSize = (FrameSize + Length) * 2 + FF_INPUT_BUFFER_PADDING_SIZE;
+	if (BufferSize < Length + FrameSize + AV_INPUT_BUFFER_PADDING_SIZE) {
+		size_t NewSize = (FrameSize + Length) * 2 + AV_INPUT_BUFFER_PADDING_SIZE;
 		safe_aligned_reallocz(Buffer, BufferSize, NewSize);
 		BufferSize = NewSize;
 	}
@@ -340,5 +340,5 @@
 	}
 
 	if (FrameSize)
-		memset(Buffer + FrameSize, 0, FF_INPUT_BUFFER_PADDING_SIZE);
+		memset(Buffer + FrameSize, 0, AV_INPUT_BUFFER_PADDING_SIZE);
 }
--- ffms2-2.20/src/core/utils.cpp~	2018-04-29 09:41:45.000000000 +0200
+++ ffms2-2.20/src/core/utils.cpp	2018-04-29 09:43:06.088381421 +0200
@@ -173,7 +173,7 @@
 	// but I don't think anyone actually uses that.
 
 	if (PrivateDataSrc && PrivateDataSize > 0) {
-		CodecContext->extradata = static_cast<uint8_t *>(av_mallocz(PrivateDataSize + FF_INPUT_BUFFER_PADDING_SIZE));
+		CodecContext->extradata = static_cast<uint8_t *>(av_mallocz(PrivateDataSize + AV_INPUT_BUFFER_PADDING_SIZE));
 		CodecContext->extradata_size = PrivateDataSize;
 		memcpy(CodecContext->extradata, PrivateDataSrc, PrivateDataSize);
 	}
